src/k8s-configuration/azext_k8s_configuration/providers/FluxConfigurationProvider.py

```diff
@@ -17,515 +23,1031 @@
-                consts.SCC_EXISTS_ON_CLUSTER_HELP)
-
-    def _validate_extension_install(self, resource_group_name, cluster_rp, cluster_type, cluster_name, no_wait):
-        # Validate if the extension is installed, if not, install it
-        extensions = self.extension_client.list(resource_group_name, cluster_rp, cluster_type, cluster_name)
-        flux_extension = None
-        for extension in extensions:
-            if extension.extension_type.lower() == consts.FLUX_EXTENSION_TYPE:
-                flux_extension = extension
-                break
-        if not flux_extension:
-            logger.warning("'Microsoft.Flux' extension not found on the cluster, installing it now."
-                           " This may take a few minutes...")
-
-            extension = Extension(
-                extension_type="microsoft.flux",
-                auto_upgrade_minor_version=True,
-                release_train=os.getenv(consts.FLUX_EXTENSION_RELEASETRAIN),
-                version=os.getenv(consts.FLUX_EXTENSION_VERSION)
-            )
-            if not is_dogfood_cluster(self.cmd):
-                extension = self.__add_identity(extension,
-                                                resource_group_name,
-                                                cluster_rp,
-                                                cluster_type,
-                                                cluster_name)
-
-            logger.info("Starting extension creation on the cluster. This might take a few minutes...")
-            sdk_no_wait(no_wait, self.extension_client.begin_create, resource_group_name, cluster_rp, cluster_type,
-                        cluster_name, "flux", extension).result()
-            # Only show that we have received a success when we have --no-wait
-            if not no_wait:
-                logger.warning("'Microsoft.Flux' extension was successfully installed on the cluster")
-        elif flux_extension.provisioning_state == consts.CREATING:
-            raise DeploymentError(
-                consts.FLUX_EXTENSION_CREATING_ERROR,
-                consts.FLUX_EXTENSION_CREATING_HELP
+
+def __add_identity(
+    cmd, extension_instance, resource_group_name, cluster_rp, cluster_type, cluster_name
+):
+    subscription_id = get_subscription_id(cmd.cli_ctx)
+    resources = cf_resources(cmd.cli_ctx, subscription_id)
+
+    if cluster_type.lower() == consts.MANAGED_CLUSTER_TYPE:
+        return extension_instance
+
+    cluster_rp, parent_api_version = get_cluster_rp_api_version(cluster_type)
+
+    cluster_resource_id = (
+        "/subscriptions/{0}/resourceGroups/{1}/providers/{2}/{3}/{4}".format(
+            subscription_id, resource_group_name, cluster_rp, cluster_type, cluster_name
+        )
+    )
+
+    try:
+        resource = resources.get_by_id(cluster_resource_id, parent_api_version)
+        location = str(resource.location.lower())
+    except HttpResponseError as ex:
+        raise ex
```

Comment:
Could you please use a specific error ?

Reply:
How would you like this to be more specific? We are already checking for one error type. If we fail for a different error reason, only then do we re-throw